firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -986,8 +993,28 @@ class SearchViewController: SiteTableViewController,
                 cell = twoLineCell
             }
         case .bookmarksAndHistory:
-            if model.shouldShowBookmarksSuggestions &&
-                model.shouldShowBrowsingHistorySuggestions {
+            if featureFlags.isFeatureEnabled(.firefoxSuggestFeature, checking: .buildAndUser) {
+                if model.shouldShowBookmarksSuggestions &&
+                    model.shouldShowBrowsingHistorySuggestions {
+                    if let site = data[indexPath.row] {
+                        configureBookmarksAndHistoryCell(
+                            twoLineCell,
+                            site.title,
+                            site.url,
+                            site.bookmarked ?? false
+                        )
+                        cell = twoLineCell
+                    }
+                } else if model.shouldShowBookmarksSuggestions {
+                    let site = bookmarkSites[indexPath.row]
+                    configureBookmarksAndHistoryCell(twoLineCell, site.title, site.url, true)
+                    cell = twoLineCell
+                } else if model.shouldShowBrowsingHistorySuggestions {
+                    let site = historySites[indexPath.row]
+                    configureBookmarksAndHistoryCell(twoLineCell, site.title, site.url)
+                    cell = twoLineCell
+                }
+            } else {
```

Comment:
_sniff sniff_ Smells like a Pyramid of Doom! lolololol is this extractable? Possibly not worth doing that, but just throwing it out there. THis is four conditional cases indented (switch + 3 if); that seems like a lot.